payment_payzen/models/payment_transaction.py

```diff
@@ -106,6 +117,61 @@ def _get_specific_rendering_values(self, processing_values):
         values['api_url'] = self.provider_id.payzen_get_form_action_url()
         return values
 
+    def _payzen_set_recurring_values(self, values, sale_order):
```

Comment:
You're calling this function with `self.sale_order_ids` (line 81)
Could it be an issue if `sale_order_ids` contains more than one record?
As you're assuming it's only one on the calls inside this function

Also, `self` is not used to reference the `payment.transaction`, maybe we can add `@api.model` to the function signature

Reply:
> You're calling this function with self.sale_order_ids (line 81)
Could it be an issue if sale_order_ids contains more than one record?
As you're assuming it's only one on the calls inside this function

[L74](https://github.com/odoo-ps/payment_payzen/blob/a79aa3e5ee3de681a78516fe800ed788b7d369db/payment_payzen/models/payment_transaction.py#L74) should make sure there's only one sale order.
I could add a `sale_order.ensure_one()` inside the method to make sure callers more strictly adhere to it.

> Also, self is not used to reference the payment.transaction, maybe we can add @api.model to the function signature

I think the method itself is not inherently a model one. Meaning it's just by chance that it doesn't reference the transaction itself, but it still sort of operates within the scope of one and it's not doing operations related to the model only, so not having `@api.model` would make it more flexible in case it gets overridden at some point in the future. IMO `@api.model` means more "this method should act on the model not its recordsets" therefore enforcing `self` will be an empty recordset, but that doesn't imply the reverse (no reference to self -> make it @api.model).

Reply:
Indeed, I missed line 74 :+1:
No need to add the ensure_one then

Seems good to me for api.model explanation, you understand better the behavior of the metho